src/Servers/Kestrel/Transport.Sockets/src/Internal/SocketAwaitableEventArgs.cs

```diff
@@ -52,13 +54,17 @@ public void OnCompleted(Action continuation)
             if (ReferenceEquals(_callback, _callbackCompleted) ||
                 ReferenceEquals(Interlocked.CompareExchange(ref _callback, continuation, null), _callbackCompleted))
             {
-                Task.Run(continuation);
+                Task.CompletedTask.ConfigureAwait(false).GetAwaiter().OnCompleted(continuation);
```

Comment:
I don't think we need the `.ConfigureAwait(false)`. ValueTaskAwaiter doesn't seem have it. And outside of KestrelServer's StartAsync()/StopAsync() methods (and things called by those methods), we generally don't use `ConfigureAwaitFalse()` in Kestrel since there's no SynchronizationContext in play.